From 80e2a77d233be5f90b0eec69315fb503d99a173e Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Michel=20D=C3=A4nzer?= <michel@tungstengraphics.com>
Date: Thu, 31 Jul 2008 16:54:33 +0200
Subject: [PATCH 18/41] EXA: Do still return FALSE if the driver PrepareCopy hook failed...

Thanks to Stuart Bennett for pointing out the problem on IRC.
(cherry picked from commit 148175fb8b365dcf00b13539b3f03ce33f3df707)
---
 exa/exa_accel.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/exa/exa_accel.c b/exa/exa_accel.c
index 6b3be11..09e5a7f 100644
--- a/exa/exa_accel.c
+++ b/exa/exa_accel.c
@@ -1247,6 +1247,8 @@ exaFillRegionTiled (DrawablePtr	pDrawable,
 	    }
 
 	    (*pExaScr->info->DoneCopy) (pPixmap);
+
+	    ret = TRUE;
 	}
 
 	exaMarkSync(pDrawable->pScreen);
@@ -1255,7 +1257,7 @@ exaFillRegionTiled (DrawablePtr	pDrawable,
 	    REGION_TRANSLATE(pScreen, pRegion, -xoff, -yoff);
     }
 
-    return TRUE;
+    return ret;
 }
 
 
-- 
1.6.0.4

